aptos-move/block-executor/src/executor.rs

```diff
@@ -1271,75 +1259,73 @@ where
         signature_verified_block: &[T],
         base_view: &S,
     ) -> BlockExecutionResult<BlockOutput<E::Output>, E::Error> {
-        let mut ret = if self.config.local.concurrency_level > 1 {
-            self.execute_transactions_parallel(
-                executor_arguments,
-                signature_verified_block,
-                base_view,
-            )
-            .map_err(BlockExecutionError::FallbackToSequential)
-        } else {
-            self.execute_transactions_sequential(
+        if self.config.local.concurrency_level > 1 {
+            let parallel_result = self.execute_transactions_parallel(
                 executor_arguments,
                 signature_verified_block,
                 base_view,
-                false,
-            )
-        };
+            );
 
-        let mut resource_group_bcs_fallback = matches!(
-            ret,
-            Err(BlockExecutionError::FallbackToSequential(PanicOr::Or(
-                IntentionalFallbackToSequential::ResourceGroupSerializationError
-            )))
-        );
+            // If parallel gave us result, return it
+            if let Ok(output) = parallel_result {
+                return Ok(output);
+            }
+
+            // All logs from the parallel execution should be cleared and not reported.
+            // Clear by re-initializing the speculative logs.
+            init_speculative_logs(signature_verified_block.len());
 
-        if self.config.local.concurrency_level > 1 && !resource_group_bcs_fallback {
-            // Sequential execution fallback, only worth doing if we did a different pass before,
-            // i.e. parallel. This fallback does not handle resource group serialization issues.
+            debug!("parallel execution requiring fallback");
+        }
 
-            if let Err(BlockExecutionError::FallbackToSequential(_)) = &ret {
-                // Any error logs are already written at appropriate levels.
-                debug!("Sequential_fallback occurred");
+        // If we didn't run parallel or it didn't finish successfully - run sequential
+        let sequential_result = self.execute_transactions_sequential(
+            executor_arguments,
+            signature_verified_block,
+            base_view,
+            false,
+        );
 
-                // All logs from the parallel execution should be cleared and not reported.
+        // If sequential gave us result, return it
+        let sequential_error = match sequential_result {
+            Ok(output) => {
+                return Ok(output);
+            },
+            Err(SequentialBlockExecutionError::ResourceGroupSerializationError) => {
+                // All logs from the first pass of sequential execution should be cleared and not reported.
                 // Clear by re-initializing the speculative logs.
                 init_speculative_logs(signature_verified_block.len());
```

Comment:
I think if there is logging specifically in the sequential execution logic, it may not be using speculative logging macros (since it never had a fallback). Let's add a TODO(clean-up): to check it later? (I'm happy to audit that at some point, pretty low pri most likely, but still)